proposals/kapp/001-crd-upgrade-safety/README.md

```diff
@@ -1,56 +1,99 @@
+This does not mean that we shouldn't add any specific validations against changed fields before a release of `kapp` containing this preflight check, but rather focus on laying a foundation in which we can iteratively build on.
+Some potential field validations we could add after the base implementation is in place are:
+
+- Type is not changed
+- Enum value is not removed
+- Minimum value is not increased
+- Maximum value is not decreased
+
+Due to the nature of this check, it is anticipated that users will want to be able to configure the behavior to best meet their needs. To start with, there will be a couple configuration options for this preflight check:
+
+- **mode** - allowed values: ["error", "warn"] - This is the mode that should be used when running the preflight check. When run in the `error` mode, this check will fail if any problems are found with upgrading a CRD and prevent the changes from being applied to the cluster. When run in `warn` mode, this check will always succeed, but log warnings for any problems found. The `warn` mode allows users to be aware of potential problems but proceed with applying the changes to the cluster if all other enabled preflight checks pass. `error` is the default.
+- **failMode** - allowed values: ["open", "closed"] - This is the failure mode that should be used when running the preflight check. Only used when `mode=error`. When running in the `open` mode, if a change to an existing field is not validated by an existing check the preflight check will pass. When running in the `closed` mode, all changes to an existing field _must_ be validated by an existing check. If it has not, the preflight check will fail with a message along the lines of "unable to determine if change to existing field is safe ...". `closed` is the default.
+
+#### Preflight Configuration
+
 The `CRDUpgradeSafety` preflight check is a very nuanced preflight check and different users will likely want to be able to configure this check a bit further to do things like:
+
 - Only issue warnings about potential problems
 - Fail closed/open
 
-In order to accomodate this configurability without introducing various versions of the same preflight check, the logic in https://github.com/carvel-dev/kapp/tree/develop/pkg/kapp/preflight will be updated to allow for configurable preflight checks using the [existing kapp configuration file](https://carvel.dev/kapp/docs/v0.60.x/config).
+In order to accomodate this configurability without introducing various versions of the same preflight check, the ability to specify preflight configurations will be added to the existing [`kapp` configuration file](https://carvel.dev/kapp/docs/v0.60.x/config/).
 
 As an example, configuring a preflight check could look something like:
+
 ```yaml
 apiVersion: kapp.k14s.io/v1alpha1
 kind: Config
-...
+---
```

Comment:
`---`?

Reply:
I'm not sure what happened here. Seems like my editor decided to do that for me :(

Reply:
Should be fixed in https://github.com/carvel-dev/carvel/pull/729/commits/28b0ff08bf33f3af591aeb8bfb2c19c69a7aaa06